lib/rspec/mocks/verifying_proxy.rb

```diff
@@ -34,23 +57,25 @@ def add_message_expectation(method_name, opts={}, &block)
         ensure_implemented(method_name)
         super
       end
+    end
 
-      def ensure_implemented(method_name)
-        return unless method_reference[method_name].unimplemented?
+    # @private
+    module MaybeVerifyProxyMethods
```

Comment:
> They were introduced to prevent having to always check, as I think the normal use case is not going to be to skip verification,

I think the cost of the check is negligible--but the cost of the added maintenance from trying to avoid the check when not needed is significant.

If we're going to take on the maintenance burden of the extra double class types and modules, I would want benchmarks showing what kind of performance benefit the added maintenance cost is buying us so we can make an informed decision.

On top of that, there's the snippet I posted above:

``` ruby
dbl = instance_double("MyClass")
without_verifying_partial_doubles do
  allow(dbl).to receive(:some_method_that_would_fail_verification)
end
```

If I was a user, I'd expect that to work, but if we implement things to avoid the checks for verifying doubles created before the `without_verifying_partial_doubles`, there is no way to make it work.

Do you not view that as a bug like I do?